Added reference particle helper functions.

cemitch99 opened the pull request:
Added the following functions associated with the reference particle in ImpactXParticleContainer.cpp:

Set_Energy_MeV - initialize reference energy
GetGamma - return reference particle gamma
GetBeta - return reference particle beta
GetBetaGamma - return beta*gamma
Get_Energy_MeV - return reference particle kinetic energy

Close #178

cemitch99 committed:
Added reference particle helper functions.

ax3l commented on src/particles/ImpactXParticleContainer.H:
```diff
@@ -155,6 +166,43 @@ namespace impactx
+         * @returns relativistic gamma
+         */
+        amrex::ParticleReal
+        GetGamma (RefPart const refpart);
```
Since we have access to the reference particle, we could make this an empty argument.

Alternatively, we could implement all setters and getters directly in the `RefPart` class? That is what I originally thought and would make it more explicit which energy/gamma/etc. we mean.

cemitch99 committed:
Moved reference particle helper functions.

ax3l committed:
`RefPart`: `const`-ify member functions

- make getters const
- rename to a style we will strive for (no camel case for
  member functions)

ax3l committed:
Python Bindings

ax3l committed:
Simplify Usage: RefParticle Init

ax3l committed:
Python simply: `ref_particle` getter
